docs/data/introduction/licensing/licensing.md

```diff
@@ -142,20 +143,70 @@ They include a dependency called `@mui/x-license-pro`, used to validate the lice
 If you're upgrading from community, you may want to check the [upgrading](#upgrading-from-community) section.
 
 With a commercial packaged installed, use `LicenseInfo` to set your licence key as in the code snippet below.
+Calling `setLicenseKey()` "install" the key.
 
 ```jsx
 import { LicenseInfo } from '@mui/x-license-pro';
 
 LicenseInfo.setLicenseKey('YOUR_LICENSE_KEY');
 ```
 
+You only need to install the key once in your application.
+
 ### Where to install the key
 
-You need to call `setLicenseKey` before React renders the first component.
-You only need to install the key once in your application.
+You need to call `setLicenseKey()` before React renders the first component.
 
-:::info
-When using Next.js, you should call `setLicenseKey` in [`_app.js`](https://nextjs.org/docs/advanced-features/custom-app):
+The bundle size of `setLicenseKey()` is relatively small, it should be small enough for you to be able to call it in all your bundles, regardless of whether a commercial MUI X component is rendered or not.
+
+#### Next.js app
+
+When using Next.js app, if have multiple options to install the license key.
+
+1. If your [`layout.js`](https://nextjs.org/docs/app/api-reference/file-conventions/layout) is already using `'use client'`, you can do:
+
+```tsx
+'use client';
+
+import { LicenseInfo } from '@mui/x-license-pro';
+
+LicenseInfo.setLicenseKey('YOUR_LICENSE_KEY');
+```
+
+2. Otherwise (recommended), you can create a dummy component called `MuiXLicense.tsx`:
+
+```tsx
+'use client';
+
+import { LicenseInfo } from '@mui/x-license-pro';
+
+LicenseInfo.setLicenseKey('YOUR_LICENSE_KEY');
+
+export default function MuiXLicense() {
+  return null;
+}
+```
+
+And render `<MuiXLicense>` in your [`layout.js`](https://nextjs.org/docs/app/api-reference/file-conventions/layout):
+
+```tsx
+import MuiXLicense from './MuiXLicense';
+
+export default function RootLayout(props: { children: React.ReactNode }) {
+  return (
+    <html lang="en">
+      <body>
+        {props.children}
+        <MuiXLicense />
```

Comment:
FYI: I had independently implemented something similar to this in my Next.js 13 appdir app to set the license key but without this no-op component, just relying on the side-effect of importing the licensing module to set the key. I'm not observing any issues in our deployed environment from this implementation approach.
Are there benefits to requiring this component in addition to the side-effect of setting the license key?

<details>
<summary>Nitty-gritty:</summary>

app/layout.tsx:

```tsx
import '../lib/mui-x/license'

// rest of layout
```

lib/mui-x/license.ts:

```tsx
'use client'

import { LicenseInfo } from '@mui/x-license-pro'

const licenseKey = process.env.NEXT_PUBLIC_MUI_X_LICENSE_KEY

if (licenseKey) {
  LicenseInfo.setLicenseKey(licenseKey)
}
```


</details>